Example/Tests/Tests.m

```diff
@@ -176,6 +176,32 @@
                                                                                   @"url" : @"seinfeld.wikia.com/wiki/The_Puffy_Shirt",
                                                                                   @"Feed Type" : @"private" } withGroups:nil outOfSession:false];
         });
+    
+        it(@"does not call screen if trackNamedPages = false", ^{
+            integration = [[SEGAmplitudeIntegration alloc] initWithSettings:@{ @"trackNamedPages" : @false } andAmplitude:amplitude andAmpRevenue:amprevenue andAmpIdentify:identify];
+
+            SEGScreenPayload *payload = [[SEGScreenPayload alloc] initWithName:@"Shirts" properties:@{} context:@{} integrations:@{}];
+            [integration screen:payload];
+            [verifyCount(amplitude, never()) logEvent:@"Viewed Shirts Screen" withEventProperties:@{}];
+        });
+    
+        it(@"does not call screen if trackNamedPages = true, but screen has no name", ^{
+            integration = [[SEGAmplitudeIntegration alloc] initWithSettings:@{ @"trackAllPages" : @true } andAmplitude:amplitude andAmpRevenue:amprevenue andAmpIdentify:identify];
+
+            SEGScreenPayload *payload = [[SEGScreenPayload alloc] initWithName:@"" properties:@{} context:@{} integrations:@{}];
+            [integration screen:payload];
+            
+            [verifyCount(amplitude, never()) logEvent:@"Viewed  Screen" withEventProperties:@{}];
+        });
+    
+    
+        it(@"trackNamedPages", ^{
+            integration = [[SEGAmplitudeIntegration alloc] initWithSettings:@{ @"trackNamedPages" : @true } andAmplitude:amplitude andAmpRevenue:amprevenue andAmpIdentify:identify];
```

Comment:
this part of code in all cases looks similar: 
` integration = [[SEGAmplitudeIntegration alloc] initWithSettings:@{ @"trackNamedPages" : @true } andAmplitude:amplitude andAmpRevenue:amprevenue andAmpIdentify:identify];
SEGScreenPayload *payload = [[SEGScreenPayload alloc] initWithName:@"Shirts" properties:@{} context:@{} integrations:@{}];
             [integration screen:payload];
             [verify(amplitude) logEvent:@"Viewed Shirts Screen" withEventProperties:@{} withGroups:nil outOfSession:false];`
Is it possible to take out the part in a method pass necessary arguments as per condition?

Reply:
I think that it's fine that these test cases are written out explicitly, that is the convention established in the other cases written in this file and there are no utility methods being defined and used in multiple cases.

Reply:
that's ok for me.